src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -347,8 +348,12 @@ public int compare(Subnetwork o1, Subnetwork o2) {
         return subnetworks;
     }
 
-    public Operation insertInstance(String projectId, Instance i) throws IOException {
-        return compute.instances().insert(projectId, i.getZone(), i).execute();
+    public Operation insertInstance(String projectId, String template, Instance i) throws IOException {
```

Comment:
If template is meant to be an optional argument, that should be made explicit with an Optional<T> type.